compiler/rustc_serialize/src/opaque.rs

```diff
@@ -55,6 +55,11 @@ macro_rules! write_leb128 {
     }};
 }
 
+/// A byte that cannot occur in UTF8 sequences. Used to mark the end of a string.
+/// This way we can skip validation and still be relatively sure that deserialization
+/// did not desynchronize.
+const STR_SENTINEL: u8 = 0xC1;
```

Comment:
This byte does indeed not seem to appear in UTF-8 -- but it took some digging to find that out and my "evidence" still consists of just not finding the value in various character tables. I'd prefer to use something here that more obviously is disallowed, like `0xFF` or `0xFE` (see https://en.wikipedia.org/wiki/UTF-8#Encoding). Or alternative provide a link to a reliable source that states that this byte cannot be part of UTF-8.

Reply:
I think it falls out of the way UTF-8 is encoded. A byte that starts with bits `110` must be the start of a 2-byte sequence (it's not valid anywhere else). The 2-byte sequence starting with the bits `11000001` maps to characters in the ASCII range (U+40 to U+7F). Therefore it's an "overlong encoding" and cannot appear in valid Unicode. https://en.wikipedia.org/wiki/UTF-8#Overlong_encodings

Reply:
It is as @ChrisDenton explained. The byte table in the wikipedia article you marks it as grey for that reason.

I added a link.

Reply:
Thanks! That's sufficient documentation for me `:)`